### PR TITLE
feat(route): add 591 租屋網

### PR DESCRIPTION
**Pull request by Yukaii**
## 该 PR 相关 Issue / Involved issue

無

## 完整路由地址 / Example for the proposed route(s)

```routes
/591/tw/rent/:query?
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [x] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [ ] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [ ] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: docs/en/other.md [hunk 1]
```diff
@@ -4,6 +4,16 @@ pageClass: routes
 
 # Uncategorized
 
+## 591 Rental house
+
+<Route author="Yukaii" example="/591/query/order=posttime&orderType=desc" path="/591/query/:query?" :paramsDesc="['Query Parameters']"/>
```
Comment (TonyRL) [inline docs/en/other.md:9]:
```suggestion
<RouteEn author="Yukaii" example="/591/query/order=posttime&orderType=desc" path="/591/query/:query?" :paramsDesc="['Query Parameters']">
```

File: docs/en/other.md [hunk 2]
```diff
@@ -4,6 +4,16 @@ pageClass: routes
...
+
+::: tip Tip
+
+Copy the URL of the 591 filter housing page and remove the front part "https://rent.591.com.tw/?", you will get the query parameters.
+
+:::
+
```
Comment (TonyRL) [inline docs/en/other.md:16]:
```suggestion

</RouteEn>

```

File: docs/other.md [hunk 3]
```diff
@@ -18,6 +18,16 @@ pageClass: routes
 
 </Route>
 
+## 591 租屋網
+
+<Route author="Yukaii" example="/591/query/order=posttime&orderType=desc" path="/591/query/:query?" :paramsDesc="['查詢參數']"/>
```
Comment (TonyRL) [inline docs/other.md:23]:
```suggestion
<Route author="Yukaii" example="/591/query/order=posttime&orderType=desc" path="/591/query/:query?" :paramsDesc="['查詢參數']">
```

File: docs/other.md [hunk 4]
```diff
@@ -18,6 +18,16 @@ pageClass: routes
...
+
+::: tip 提示
+
+複製 591 篩選房屋頁面的網址，並去掉前段 `https://rent.591.com.tw/?` 部分，即可得到查詢參數。
+
+:::
+
```
Comment (TonyRL) [inline docs/other.md:30]:
```suggestion

</Route>

```

File: lib/v2/591/list.js [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+const { CookieJar } = require('tough-cookie');
+const cheerio = require('cheerio');
+
+const cookieJar = new CookieJar();
+
+const client = got.extend({
+    cookieJar,
+    headers: {
+        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.114 Safari/537.36',
+    },
```
Comment (TonyRL) [inline lib/v2/591/list.js:11]: RSSHub will use Chrome on Mac UA by default.
https://github.com/DIYgod/RSSHub/blob/3195eb86f342aaa5ca978092633d1dea8ded465b/lib/config.js#L45

Is there any specific to use this UA over the default one?

File: lib/v2/591/list.js [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+                <div id="item-is-combine">${house.is_combine}</div>
+                <div id="item-hurry">${house.hurry}</div>
+                <div id="item-is-socail">${house.is_socail}</div>
+                <div id="item-surrounding">${house.surrounding}</div>
+                <div id="item-discount-price-str">${house.discount_price_str}</div>
+                <div id="item-cases-id">${house.cases_id}</div>
+                <div id="item-is-video">${house.is_video}</div>
+                <div id="item-preferred">${house.preferred}</div>
+                <div id="item-cid">${house.cid}</div>
+            </div>`;
```
Comment (TonyRL) [inline lib/v2/591/list.js:196]: Please [use art-template](https://docs.rsshub.app/joinus/script-standard.html#template)
docs: https://aui.github.io/art-template/docs/syntax.html

File: lib/v2/591/list.js [hunk 7]
```diff
@@ -0,0 +1,211 @@
...
+
+        return {
+            title: itemTitle,
+            description,
+            link: itemUrl,
+        };
+    });
+
+    ctx.state.data = {
+        title: `591 租屋 - 自訂查詢`,
```
Comment (TonyRL) [inline lib/v2/591/list.js:206]:
```suggestion
        title: '591 租屋 - 自訂查詢',
```

File: docs/en/other.md (hunk 1 above)
Comment (TonyRL) [inline docs/en/other.md:9]: It is suggested to add `tw` in `path` to avoid confusion between `rent.591.com.hk` and `rent.591.com.tw`

**Review by TonyRL — CHANGES_REQUESTED** [review]

File: lib/v2/591/list.js (hunk 5 above)
Comment (Yukaii) [reply lib/v2/591/list.js:11]: Thanks! I didn't know that, I just move my code from other places here. I'll test it to see if it works. 👍

File: lib/v2/591/list.js (hunk 7 above)
Comment (Yukaii) [reply lib/v2/591/list.js]: Updated in 9974503f507003548b372be361cfbbc4e4349da3

File: docs/en/other.md (hunk 1 above)
Comment (Yukaii) [reply docs/en/other.md]: fixed in bb9c3b6

File: lib/v2/591/list.js (hunk 6 above)
Comment (Yukaii) [reply lib/v2/591/list.js]: fixed in f607581

File: lib/v2/591/templates/house.art [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+
+<div id="more-pictures">
+    {{each photoList}}
+        <img src="{{$value}}" style="margin-bottom: 20px;">
+    {{/each}}
+</div>
+
+<div id="raw-data" style="display: none;">
+    {{rawHouse}}
+</div>
```
Comment (TonyRL) [inline lib/v2/591/templates/house.art:56]: Please use ctx.state.json instead
docs: https://docs.rsshub.app/joinus/script-standard.html#ctx-state-json

Comment (Yukaii) [reply lib/v2/591/templates/house.art]: Thank you! Fixed

File: lib/v2/591/list.js [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+*/
+/**
+
+@typedef {object} Surrounding
+@property {string} type - The type of the surrounding.
+@property {string} desc - The description of the surrounding.
+@property {string} distance - The distance to the surrounding.
+*/
+
+const renderHouse = (house) => art(path.join(__dirname, 'templates/house.art'), { house, rawHouse: JSON.stringify(house) });
```
Comment (TonyRL) [inline lib/v2/591/list.js:104]:
```suggestion
const renderHouse = (house) => art(path.join(__dirname, 'templates/house.art'), { house });
```

**Pull request merged**